Patch-Source: https://github.com/nginx/unit/commit/919cae7ff95a3ce5878731a8d23f34c75489d3b4
From 919cae7ff95a3ce5878731a8d23f34c75489d3b4 Mon Sep 17 00:00:00 2001
From: Andrew Clayton <a.clayton@nginx.com>
Date: Wed, 15 Nov 2023 03:34:49 +0000
Subject: [PATCH] PHP: Fix a possible file-pointer leak.

In nxt_php_execute() it is possible we could bail out before cleaning up
the FILE * representing the PHP script to execute.

At this point we only need to call fclose(3) on it.

We could have possibly moved the opening of this file to later in the
function, but it is probably good to bail out as early as possible if we
can't open it.

This was found by Coverity.

Fixes: bebc03c72 ("PHP: Implement better error handling.")
Signed-off-by: Andrew Clayton <a.clayton@nginx.com>
---
 src/nxt_php_sapi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/nxt_php_sapi.c b/src/nxt_php_sapi.c
index ba000fc02..77117533f 100644
--- a/src/nxt_php_sapi.c
+++ b/src/nxt_php_sapi.c
@@ -1225,6 +1225,8 @@ nxt_php_execute(nxt_php_run_ctx_t *ctx, nxt_unit_request_t *r)
         nxt_unit_req_debug(ctx->req, "php_request_startup() failed");
 
         nxt_unit_request_done(ctx->req, NXT_UNIT_ERROR);
+        fclose(fp);
+
         return;
     }
 
